internal/services/network/network_watcher_flow_log_resource.go

```diff
@@ -182,18 +182,16 @@ func resourceNetworkWatcherFlowLogCreateUpdate(d *pluginsdk.ResourceData, meta i
 
 	// guaranteed via schema validation
 	nsgId, _ := parse.NetworkSecurityGroupID(networkSecurityGroupID)
-	id := parse.NewFlowLogID(subscriptionId, resourceGroupName, networkWatcherName, *nsgId)
 
-	locks.ByID(nsgId.ID())
-	defer locks.UnlockByID(nsgId.ID())
+	id := parse.NewFlowLogIDShim(subscriptionId, resourceGroupName, networkWatcherName, d.Get("name").(string), *nsgId)
```

Comment:
how come we are removing the lock?

Reply:
Sorry for that, I've added them back.
Besides, do you think we should support the way how to import the resource backward compatibly? Or I should just add a ID migration and use the new way to import only?

Reply:
we'll need to do a state migration, could you update this PR with the new 3.0 beta "this is for good" flag so this can be included in the beta?